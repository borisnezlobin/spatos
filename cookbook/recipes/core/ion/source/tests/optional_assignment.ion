let a ?= 5
echo $a
let a ?= 10
echo $a

let b ?= [ 1 2 3 ]
echo @b
let b ?= [ 4 5 6 ]
echo @b

let c d e ?= [ 1 2 3 ] hello 5
echo @c
echo $d
echo $e
let c d e ?= [ 4 5 6 ] world 10
echo @c
echo $d
echo $e
