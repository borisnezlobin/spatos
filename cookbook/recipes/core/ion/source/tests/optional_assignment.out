5
5
1 2 3
1 2 3
1 2 3
hello
5
1 2 3
hello
5
